src/Compilers/Server/VBCSCompilerTests/DesktopBuildClientTests.cs

```diff
@@ -160,24 +161,17 @@ public void ConnectToServerFails()
                                                                                          cts.Token),
                                           cts.Token);
                 Assert.False(connection.IsCompleted);
-                // Spin for a little bit
-                await Task.Delay(TimeSpan.FromMilliseconds(100));
                 cts.Cancel();
-                await Task.WhenAny(connection, Task.Delay(TimeSpan.FromMilliseconds(100)))
-                    .ConfigureAwait(false);
-                Assert.True(connection.IsCompleted);
-                Assert.True(connection.IsCanceled);
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(
+                    async () => await connection.ConfigureAwait(false)).ConfigureAwait(false);
```

Comment:
I tried that, but it appeared that the exception was swallowed in the task object, so it was never witnessed by the call.